### PR TITLE
Drop support for quarkus-smallrye-metrics

### PR DESCRIPTION
**Pull request by fedinskiy**
https://issues.redhat.com/browse/QUARKUS-4144

### CODE REVIEW

**Comment by fedinskiy** [comment]
I disabled a test, which tries to update to 3.9, until we will be able to update the rest of the TS to 3.9

**Comment by fedinskiy** [comment]
CodeQuarkusTest uses the latest version from code.quarkus (3.9 at the moment of writing), so its failure is not related

File: testsuite/src/it/java/io/quarkus/ts/startstop/QuarkusMavenPluginTest.java [hunk 1]
```diff
@@ -96,6 +96,7 @@ public void infoTarget(TestInfo testInfo) throws IOException, InterruptedExcepti
 
     @Test
     @DisabledOnOs({OS.WINDOWS}) // https://github.com/quarkusio/quarkus/issues/33403
+    @Disabled("https://github.com/quarkus-qe/quarkus-startstop/issues/356")
```
Comment (rsvoboda) [inline testsuite/src/it/java/io/quarkus/ts/startstop/QuarkusMavenPluginTest.java:99]: There is no description in the issue

And there is https://github.com/quarkus-qe/quarkus-startstop/pull/353 in progress

I would drop this disable line

**Review by michalvavrik — COMMENTED** [review]
What about:

- https://github.com/quarkus-qe/quarkus-startstop/blob/455e90138a530de78a8f3fc203ebbc6affe91028/testsuite/src/it/java/io/quarkus/ts/startstop/utils/CodeQuarkusExtensions.java#L154

- https://github.com/quarkus-qe/quarkus-startstop/blob/455e90138a530de78a8f3fc203ebbc6affe91028/app-full-microprofile/pom.xml#L41

**Comment by fedinskiy** [comment]
@michalvavrik I tried that, but we use them, and micrometer doesn't support microprofile yet[1], so we can not simply move 

[1] "Use the SmallRye Metrics extension when it is required to retain MicroProfile specification compatibility." https://quarkus.io/guides/smallrye-metrics

**Review by michalvavrik — APPROVED** [review]
Thanks for the explanation.

**Pull request merged**